backend/inspirehep/utils.py

```diff
@@ -174,3 +174,48 @@ def _exit_handler(record_uuids, task_name, signum, frame):
 def setup_celery_task_signals(record_ids, task_name):
     signal.signal(signal.SIGINT, partial(_exit_handler, record_ids, task_name))
     signal.signal(signal.SIGTERM, partial(_exit_handler, record_ids, task_name))
+
+
+def set_value_by_path(o, k, v, accessor: callable = lambda o, k: o.setdefault(k, dict()) if hasattr(o, "setdefault") else o[k],
+             getter: callable = None, setter: callable = None, sep: str = '.'):
+    keys = __keys(k, sep)
+    getter = __getter(getter, accessor)
+    setter = __setter(setter)
+    for i in range(len(keys) - 1):
+        if o is None:
+            return 0
+        o = getter(o, keys[i])
+
+    return setter(o, keys[-1], v)
+
+
+def __keys(key, sep: str):
+    if isinstance(key, str):
+        return key.split(sep=sep)
+    else:
+        return list(key)
+
+
+def __getter(getter: callable, accessor: callable):
+    def __default_getter(o, k):
+        if isinstance(o, list):
+            return [accessor(i, k) for i in o]
+        else:
+            return accessor(o, k)
+
+    return __default_getter if getter is None else getter
```

Comment:
@nooraangelva is this copy-paste from somewhere else?

Reply:
Yes, from this repository -> https://github.com/mbello/dict-deep/tree/7c05652c5c845886ea3b36d2fb8b95d5618e1652

Reply:
I would suggest installing it as a package https://pypi.org/project/dict-deep/ :)

Reply:
Much better idea. I should have asked if I could install it. 😄 My last internship place didn't really like open source packages.